src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -336,7 +338,7 @@ internal Location CreateLocation(LocationVersionOne v1Location)
             {
                 location = new Location
                 {
-                    FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
+                    FullyQualifiedLogicalName = v1Location.FullyQualifiedLogicalName ?? v1Location.LogicalLocationKey,
```

Comment:
>FullyQualifiedLogicalName [](start = 59, length = 25)

This isn't right. According to the spec (I looked as far back as the original v1 "working draft" from which the TC started its work), `LogicalLocationKey` is never guaranteed to be a valid FQLN. It is only guaranteed to match one of the property names in the `run.logicalLocations` dictionary.

If `v1.FQLN` is missing, your only choice is to omit `v2.FQLN`, so the correct code here is:
```
location = new Location
{
    FullyQualifiedLogicalName = v1Location.FullyQualifiedLogicalName, // Works whether or not it's present in the v1 file
    ...
```

It doesn't even help to use the key to find the `logicalLocation` object in `run.logicalLocations` dictionary. In v1, a `logicalLocation` object doesn't have a `fullyQualifiedName` property, so you'd have to walk the parent chain and glue the FQLN together yourself (and you'd have to know the language syntax to do it).

So there's no good answer. Just copy the FQLN from v1 to v2 and call it a day.

Reply:
I think I might be contradicting what I commented in your previous PR. If so, my previous comment was wrong. You just can't get away with using a key as an FQLN.

---
In reply to: [244237410](https://github.com/Microsoft/sarif-sdk/pull/1185#discussion_r244237410) [](ancestors = 244237410)

Reply:
Yes, this contradicts what we discussed previously but what you say here makes perfect sense.

---
In reply to: [244237488](https://github.com/Microsoft/sarif-sdk/pull/1185#discussion_r244237488) [](ancestors = 244237488,244237410)